app/src/main/java/ai/brokk/gui/WorkspaceChip.java

```diff
@@ -560,47 +557,74 @@ protected void bindComputed() {
+            return;
+        }
+
         ContextFragment fragment = getPrimaryFragment();
         if (fragment == null) {
             return;
         }
-        updateTextAndTooltip(fragment);
-        applyTheme();
+
+        TextAndTooltip tt = getTextAndTooltip(fragment);
+
+        // Apply Swing updates on the EDT only.
+        SwingUtilities.invokeLater(() -> {
+            label.setText(tt.text());
+            label.setToolTipText(tt.tooltipHtml());
+            label.getAccessibleContext().setAccessibleDescription(tt.accessibleDescription());
+            applyTheme();
+        });
     }
 
-    protected void updateTextAndTooltip(ContextFragment fragment) {
+    protected TextAndTooltip getTextAndTooltip(ContextFragment fragment) {
+        // label
         String newLabelText;
         if (kind == ChipKind.SUMMARY) {
             // Base WorkspaceChip is not used for summaries; SummaryChip overrides this.
             newLabelText = fragment.shortDescription();
         } else if (kind == ChipKind.OTHER) {
-            String sd;
-            try {
-                sd = fragment.shortDescription();
-            } catch (Exception e) {
-                logger.warn("Unable to obtain short description from {}!", fragment, e);
-                sd = "<Error obtaining description>";
-            }
-            newLabelText = capitalizeFirst(sd);
+            newLabelText = capitalizeFirst(fragment.shortDescription());
         } else {
-            String sd;
-            try {
-                sd = fragment.shortDescription();
-            } catch (Exception e) {
-                logger.warn("Unable to obtain short description from {}!", fragment, e);
-                sd = "<Error obtaining description>";
-            }
+            var sd = fragment.shortDescription();
             newLabelText = sd.isBlank() ? label.getText() : sd;
         }
-        label.setText(newLabelText);
 
-        try {
-            label.setToolTipText(buildDefaultTooltip(fragment));
-            label.getAccessibleContext().setAccessibleDescription(fragment.description());
-        } catch (Exception ex) {
-            logger.debug("Failed to refresh chip tooltip for fragment {}", fragment, ex);
+        // tooltip
+        String descriptionSource;
+        if (fragment instanceof ContextFragment.ComputedFragment cf) {
+            descriptionSource = cf.computedDescription().renderNowOr("");
+        } else {
+            descriptionSource = fragment.description();
+        }
+        String description = StringEscapeUtils.escapeHtml4(descriptionSource)
+                .replace("\r\n", "\n")
+                .replace("\r", "\n")
+                .replace("\n", "<br/>");
+
+        String metrics = "";
+        if (fragment.isText() || fragment.getType().isOutput()) {
+            String text;
+            if (fragment instanceof ContextFragment.ComputedFragment cf) {
+                text = cf.computedText().renderNowOr("");
```

Comment:
Wondering if we shouldn't also bind a UI update on the text if we hit the "blank string" outcome